### PR TITLE
improve error message when type have same name

### LINKED ISSUE #6334: Error messages are vague when types have same name
Found through @LPGhatguy on Gitter at [January 3, 2016 10:09 PM](https://gitter.im/Microsoft/TypeScript?at=5689e26835e1a316162f379f).

Seems like #1419 didn't have enough coverage. Try out the following:
- `m1.ts`
  
  ``` ts
  export interface A { a1; }
  ```
- `m2.ts`
  
  ``` ts
  export interface A { a2; }
  ```
- `m3.ts`
  
  ``` ts
  import * as m1 from "./m1";
  import * as m2 from "./m2";
  
  let x: m1.A;
  let y: m2.A;
  
  x = y;
  x === y;
  ```

Error messages:

```
Type 'A' is not assignable to type 'A'.
  Property 'a1' is missing in type 'A'.

Operator '===' cannot be applied to types 'A' and 'A'.
```

This is the worst error message you can give to a user.

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #6334 

the part of the original issue has been fixed

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11137,12 +11147,7 @@ namespace ts {
             }
 
             function reportRelationError(message: DiagnosticMessage | undefined, source: Type, target: Type) {
-                let sourceType = typeToString(source);
-                let targetType = typeToString(target);
-                if (sourceType === targetType) {
-                    sourceType = typeToString(source, /*enclosingDeclaration*/ undefined, TypeFormatFlags.UseFullyQualifiedType);
-                    targetType = typeToString(target, /*enclosingDeclaration*/ undefined, TypeFormatFlags.UseFullyQualifiedType);
-                }
+                const [ sourceType, targetType ] = getRelationTypeName(source, target);
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Please change formatting:
```
const [sourceType, targeType] =
```

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -21838,8 +21843,7 @@ namespace ts {
             }
 
             function reportOperatorError() {
-                const leftStr = typeToString(leftType);
-                const rightStr = typeToString(rightType);
+                const [ leftStr, rightStr ] = getRelationTypeName(leftType, rightType);
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Please change formatting:
```
const [sourceType, targeType] =
```

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -3097,6 +3097,16 @@ namespace ts {
             return result;
         }
 
+        function getRelationTypeName(left: Type, right: Type): [string, string] {
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Suggested name: `getTypeNamesForErrorDisplay`

**Comment by Kingwl** [comment]
⬆️

**Comment by RyanCavanaugh** [comment]
@kingwl 🥇 for your patience

**Pull request merged**